quiche/src/recovery/gcongestion/bbr2.rs

```diff
@@ -464,6 +490,18 @@ impl BBRv2 {
                 self.cwnd,
                 self.mode.min_rtt(),
             );
+
+            if let Some(pacing_rate) =
+                self.params.initial_pacing_rate_bytes_per_second
+            {
+                // Do not allow the pacing rate calculated from the first RTT
+                // measurement to be higher than the configured initial pacing
+                // rate.
```

Comment:
Added this comment explaining the second meaning of the initial_pacing_rate_bytes_per_second parameter: serving as a max initial pacing rate

Reply:
I do believe this code only runs once per connection on the assumption that acks are always have a non zero number of bytes associated with them.  The comments in the code do suggest that this block only runs very early in the connection, before there are changes to the initial window used at the beginning of the connection.

Reply:
Consider that `total_bytes_acked` should be the `sum of all bytes_acked` on a QUIC connection path.

`self.mode.total_bytes_acked() == bytes_acked` should only happen when processing an ACK where the previous state of total_bytes_acked was 0.

Reply:
> // After the first ACK, cwnd is still the initial congestion window.

Yep comment from [here](https://github.com/cloudflare/quiche/pull/2115/files#diff-ad4b2a983ab827f21b0de5b2449da3abde4e40ef3415342499f12322e92b2bd9R488) suggests its run only once.. just wish the code was easier to reason about :\

Thx for looking into this, lets move forward with this.